django/applications/catmaid/static/js/layers/pixi-image-block-layer.js

```diff
@@ -0,0 +1,310 @@
+            .then(block => {
+              if (!CATMAID.tools.arraysEqual(this._tilesBuffer[i][j], coord)) return;
+
+              let slice = block.pick(null, null, blockZ);
+
+              if (slice.shape[0] < this.tileSource.tileWidth ||
+                  slice.shape[1] < this.tileSource.tileHeight) {
+                let empty = nj.zeros(
+                    [this.tileSource.tileWidth, this.tileSource.tileHeight],
+                    slice.dtype);
+                empty.selection.data.fill(this.fillValue);
+                var sub = empty.hi(slice.shape[0], slice.shape[1]);
+
+                for(let i=0; i<slice.shape[0]; ++i) {
+                  for(let j=0; j<slice.shape[1]; ++j) {
+                    empty.set(i,j, slice.get(i, j));
+                  }
+                }
+              }
+
+              // The array is still column major, so transpose to row-major for tex.
+              slice = slice.transpose(1, 0);
+
+              let texture = this._sliceToTexture(slice);
+              this._tilesBuffer[i][j] = [coord, texture];
+            });
+        })).then(this._swapBuffers.bind(this, false, undefined));
+        loading = true;
+      } else if (!loading) {
+        this._oldZoom = this._swapZoom;
+        this._oldZ    = this._swapZ;
+        this._renderIfReady();
+      }
+
+      if (typeof completionCallback !== 'undefined') {
+        if (loading && blocking) {
+          this._completionCallback = completionCallback;
+        } else {
+          this._completionCallback = null;
+          completionCallback();
+        }
+      }
+    }
+
+    _dtypeWebGLParams(dtype) {
+      const gl = this._context.renderer.gl;
+      var format, type, internalFormat, jsArrayType;
+
+      switch (dtype) {
+        case 'uint8':
+          format = gl.LUMINANCE;
+          type = gl.UNSIGNED_BYTE;
+          internalFormat = gl.LUMINANCE;
+          jsArrayType = Uint8Array;
+          break;
+        case 'uint32':
+          // format = gl.RGBA_INTEGER;
+          // type = gl.UNSIGNED_BYTE;
+          // internalFormat = gl.RGBA8UI;
+          // jsArrayType = Uint8Array;
+          // format = gl.RED_INTEGER;
+          // type = gl.UNSIGNED_INT;
+          // internalFormat = gl.R32UI;
+          // jsArrayType = Uint32Array;
+          format = gl.RGBA;
+          type = gl.UNSIGNED_BYTE;
+          internalFormat = gl.RGBA;
+          jsArrayType = Uint8Array;
+          break;
+        default:
+          throw new CATMAID.Error(`Unknown data type for stack layer: ${dtype}`);
+      }
+
+      return {format, type, internalFormat, jsArrayType};
+    }
+
+    _sliceToTexture(slice) {
+      let renderer = this._context.renderer;
+      let gl = renderer.gl;
+
+      let {format, type, internalFormat, jsArrayType} = this._dtypeWebGLParams(slice.dtype);
+
+      let texture = new PIXI.glCore.GLTexture(
+          gl,
+          slice.shape[1],
+          slice.shape[0],
```

Comment:
I suppose `slice.shape[1]` = width and `slice.shape[0]` = height, because you transposed the slice data on line 172?

Reply:
Oops, commented on lines changed in a later commit.